core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/LimsUtils.java

```diff
@@ -668,8 +668,7 @@ public static boolean isSolidRun(Run run) {
   public static <T extends DetailedSample> T getParent(Class<T> targetParentClass, DetailedSample start) {
     for (DetailedSample current = deproxify(start.getParent()); current != null; current = deproxify(current.getParent())) {
       if (targetParentClass.isInstance(current)) {
-        @SuppressWarnings("unchecked")
-        T result = (T) current;
+        T result = targetParentClass.cast(current);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)